contracts/facets/EmbalmerFacet.sol

```diff
@@ -216,9 +220,20 @@ contract EmbalmerFacet {
                 revert ArchaeologistListContainsDuplicate(selectedArchaeologists[i].archAddress);
             }
 
-            // todo: check convenience structure for public keys that have already been used
+            if(s.publicKeyToArchaeologistAddress[selectedArchaeologists[i].publicKey] != address(0)) {
```

Comment:
Need to make sure we have integration tests for new logic like this but I can add that on the other branch